src/Components/Firestore.js

```diff
@@ -432,3 +434,44 @@ export async function deleteWatchlistDataEntry(userId, listId) {
   // Run this script periodically from server as deleting collections
   // from a web client is not recommended by firestore.
 }
+
+export async function GetUserRxStateFromFirestore(userId, rxnTarget) {
+  const docRef = doc(db, "users", userId, "reactions", rxnTarget.entityId);
+  const docSnap = await getDoc(docRef);
+  if (!docSnap.exists()) {
+    return getDefaultUserRxns(rxnTarget);
+  }
+  return docSnap.data();
+}
+
+export async function SaveUserRxStateToFirestore(
+  userId,
+  rxnTarget,
+  rxnType,
+  userRxns,
+  incrementBy
+) {
+  const docRef = doc(db, "users", userId, "reactions", rxnTarget.entityId);
+  const countDocRef = doc(db, rxnTarget.countDocPath);
+  const countFieldPath = `reactionCounts.${rxnType.key}`;
```

Comment:
I _think_ we should be able to order by fields in a map, if we use dot notation to reference the field name, similar to what I have to do here to update one field of a map without wiping the others.  Something like `.orderBy('reactionCounts.love', descending: true)`.  This is just based on [this answer](https://stackoverflow.com/questions/70047080/flutter-firestore-how-i-can-order-firestore-query-by-field-value-inside-map), though, I haven't tried it to be 100%.  

If that works, the downside of this approach is still all the dot notation stuff, with the win of organization in the document and making it possible to build a 'default' reactionCounts object .  I'm open to putting the counts root-level -- if so, I'll probably need to add another field to a `RxnTarget` for the `countKey` so they can have a key name like `applauseCount`, similar to how you had it, since a root-level field of `love` or `trash` might be ambiguous or maybe even collide with something.  Let me know which way you prefer.

Reply:
Good find! It looks like it is possible - who knew!?! 

Lets proceed with the `reactionCount` structure as you have it. You can disregard my original comment.

Reply:
Awesome, sounds good! I just pushed some commits to address the issues you found, and add notification support for reactions this way.  Take a look and lmk what you think